src/tui/tabbed.rs

```diff
@@ -523,6 +523,38 @@ impl Tabbed {
         self.tabs[self.active_idx].set_style(TabStyle::Normal);
     }
 
+    pub fn switch(&mut self, string: &str) {
+        for (tab_idx,tab) in self.tabs.iter().enumerate() {
+            match tab.src {
+                MsgSource::Serv { ref serv_name } => {
+                    if serv_name.contains(string) {
+                        self.active_idx = tab_idx;
```

Comment:
You need to use `self.select_tab()`, do not manually set `self.active_idx` because it breaks horizontal scrolling in the tab line.

Same for other `self.active_idx = tab_idx;` lines.